lib/presentation/core/app_widget.dart

```diff
@@ -16,7 +16,7 @@ class AppWidget extends StatelessWidget {
     return MultiBlocProvider(
       providers: [
         BlocProvider<AuthBloc>(
-          create: (_) => getIt<AuthBloc>(),
+          create: (_) => getIt<AuthBloc>()..add(AuthEvent.observeUser()),
```

Comment:
Maybe we can simply start observing the user in the constructor of the AuthBloc (so when initialized).

Then we don't need an event for it.

Reply:
Previously I did call observeUser from constructor, but it throws error on github action analyze job.
Since observeUser will emit the unauthenticated state, it says
```The member 'emit' can only be used within 'package:bloc/src/bloc.dart'```

I figured we need event to avoid this. I'll look into it if there are any workaround for this.

Reply:
You can have an initial event, and in that initial event put the observeUser